hkdf.go

```diff
@@ -109,6 +109,9 @@ func (c *hkdf1) Read(p []byte) (int, error) {
 	return n, nil
 }
 
+// hkdfAllZerosSalt is a preallocated buffer of zeros used in ExtractHKDF()
+var hkdfAllZerosSalt [64]byte
+
 func ExtractHKDF(h func() hash.Hash, secret, salt []byte) ([]byte, error) {
```

Comment:
Might be a good time to add a func comment, and describe the len 0 behavior in there?